tests/SpecTests/ClientSideEncryptionSpecTest.php

```diff
@@ -395,65 +482,7 @@ public function testBSONSizeLimitsAndBatchSplitting()
 
         $document = json_decode(file_get_contents(__DIR__ . '/client-side-encryption/limits/limits-doc.json'), true);
 
-        // Test 1
-        $collection->insertOne(['_id' => 'over_2mib_under_16mib', 'unencrypted' => str_repeat('a', 2097152)]);
-
-        // Test 2
-        $collection->insertOne(
-            ['_id' => 'encryption_exceeds_2mib', 'unencrypted' => str_repeat('a', 2097152 - 2000)] + $document
-        );
-
-        // Test 3
-        $commands = [];
-        (new CommandObserver())->observe(
-            function () use ($collection) {
-                $collection->insertMany([
-                    ['_id' => 'over_2mib_1', 'unencrypted' => str_repeat('a', 2097152)],
-                    ['_id' => 'over_2mib_2', 'unencrypted' => str_repeat('a', 2097152)],
-                ]);
-            },
-            function ($command) use (&$commands) {
-                $commands[] = $command;
-            }
-        );
-
-        $this->assertCount(2, $commands);
-        foreach ($commands as $command) {
-            $this->assertSame('insert', $command['started']->getCommandName());
-        }
-
-        // Test 4
-        $commands = [];
-        (new CommandObserver())->observe(
-            function () use ($collection, $document) {
-                $collection->insertMany([
-                    [
-                        '_id' => 'encryption_exceeds_2mib_1',
-                        'unencrypted' => str_repeat('a', 2097152 - 2000),
-                    ] + $document,
-                    [
-                        '_id' => 'encryption_exceeds_2mib_2',
-                        'unencrypted' => str_repeat('a', 2097152 - 2000),
-                    ] + $document,
-                ]);
-            },
-            function ($command) use (&$commands) {
-                $commands[] = $command;
-            }
-        );
-
-        $this->assertCount(2, $commands);
-        foreach ($commands as $command) {
-            $this->assertSame('insert', $command['started']->getCommandName());
-        }
-
-        // Test 5
-        $collection->insertOne(['_id' => 'under_16mib', 'unencrypted' => str_repeat('a', 16777216 - 2000)]);
-
-        // Test 6
-        $this->expectException(BulkWriteException::class);
-        $this->expectExceptionMessageRegExp('#object to insert too large#');
-        $collection->insertOne(['_id' => 'encryption_exceeds_16mib', 'unencrypted' => str_repeat('a', 16777216 - 2000)] + $document);
+        $test($this, $collection, $document);
```

Comment:
I assume there's no problem for the closures that omit a third argument?

Reply:
This might become a problem in PHP 8, in which case my next PR will fix that part. PHP 8 will require some more work, but I wanted to extract this to keep the diffs concise.

Reply:
ArgumentCountError? Yes, I am familiar with that :rofl: